src/tiled/objectselectiontool.cpp

```diff
@@ -825,6 +825,24 @@ void ObjectSelectionTool::updateHandleVisibility()
     for (int i = 0; i < AnchorCount; ++i)
         mResizeHandles[i]->setVisible(showHandles && mMode == Resize);
 
+    // Hide extra handles for the straight line polygons.
+    foreach (MapObject *mapObj, mapDocument()->selectedObjects()) {
+        if (mapObj->shape() == MapObject::Polyline &&
+            mapObj->polygon().size() == 2) {
+            mResizeHandles[TopLeftAnchor]->setVisible(false);
+            mResizeHandles[BottomLeftAnchor]->setVisible(false);
+            mResizeHandles[TopRightAnchor]->setVisible(false);
+            mResizeHandles[BottomRightAnchor]->setVisible(false);
```

Comment:
Hmm, this seems wrong to me. You're hiding all corner anchors as soon as you see a polyline with only two points. However, there can be a diagonal polyline with two points, for which these handles make perfect sense.

Another problem with the patch is that it doesn't fix the same problem for polylines with more than two points that all line up either vertically or horizontally. This means the check should probably be done on the bounding box.

Finally, you're going over all selected objects and discard some handles as soon as one of them would be a short polyline. However, when there is a short polyline selected in combination with some other object, or even just two short polylines in different positions, then all handles make sense and work fine.

Now, it's too late for me to consider the correct fix. Feel free to have another go at it, but otherwise I will look into it later.